generator/src/main/java/com/algorand/sdkutils/listeners/TemplateGenerator.java

```diff
@@ -0,0 +1,256 @@
+
+            Set<String> types = new HashSet<>();
+            for (TypeDef typeDef: model.getValue()) {
+                types.add(typeDef.rawTypeName);
+                if (typeDef.isOfType("array")) {
+                    types.add("array");
+                }
+            }
+
+            // TODO: Arg to inject model package.
+            context.put("package", "com.algorand.algosdk.v2.client.model");
+            context.put("def", model.getKey());
+            context.put("props", model.getValue());
+            context.put("types", types);
+
+            StringWriter writer = new StringWriter();
+            template.merge( context, writer );
+
+            // TODO: Write to files instead of stdout.
+            System.out.println("====================");
+            System.out.println(StringUtils.capitalize(model.getKey().name) + ".java");
+            System.out.println("====================");
+            System.out.println(writer.toString());
+        }
+    }
+
+    private void writeQueryClass(Template t) {
+        for (QueryDef query : queries) {
+            VelocityContext context = getContext();
+
+            // Make it easier to loop thorugh parameters without caring what they are.
+            List<TypeDef> parameters = new ArrayList<>();
+            parameters.addAll(query.queryParameters);
+            parameters.addAll(query.pathParameters);
+            parameters.addAll(query.bodyParameters);
+
+            // Make it easier to get the types
+            Set<String> types = new HashSet<>();
+            for (TypeDef typeDef: parameters) {
+                types.add(typeDef.rawTypeName);
+                if (typeDef.isOfType("array")) {
+                    types.add("array");
+                }
+                if (typeDef.isOfType("enum")) {
+                    types.add("enum");
+                }
+            }
+            if (query.returnType != "String") {
+                types.add(query.returnType);
+            }
+
+            List<String> pathParts = Stream.of(query.path.split("/"))
+                            .filter(part -> !part.equals(""))
+                            .collect(Collectors.toList());
+
+            context.put("params", parameters);
+            context.put("path", pathParts);
+            context.put("types", types);
+            // TODO: Arg to inject client package.
+            context.put("package", "com.algorand.algosdk.v2.client.algod");
+
+            context.put("q", query);
+
+            StringWriter writer = new StringWriter();
+            t.merge( context, writer );
+
+            // TODO: Write to files instead of stdout.
+            System.out.println("====================");
+            System.out.println(StringUtils.capitalize(query.name) + ".java");
+            System.out.println("====================");
+            System.out.println(writer.toString());
+        }
+    }
+
+    @Override
+    public void terminate() {
```

Comment:
The template magic entrypoint. terminate is called after OpenAPI parsing completes at which point the template functions for models / queries / client is called. They all work slightly differently.

I'm still working on getting all of the parameters injected (i.e. the package).
Also a couple higher level knobs are needed for configuring filenames, and whether or not models are written 1 model per file, or all models in one file.